rasa/graph_components/providers/domain_without_response_provider.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+from typing import Dict, Text, Any
+import copy
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import Domain
+
+
+class DomainWithoutResponseProvider(GraphComponent):
+    """Provides domain without information about responses."""
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> DomainWithoutResponseProvider:
+        """Creates component (see parent class for full docstring)."""
+        return cls()
+
+    def recreate_without_responses(self, domain: Domain) -> Domain:
+        """Recreates the given domain but acts as if responses have not been specified.
+
+        Args:
+            domain: A domain.
+
+        Returns:
+            Domain that has been created from the same parameters as the given domain
+            but with an empty set of responses.
+        """
+        responses = dict()
+        domain = copy.deepcopy(domain)
+        domain.responses = responses
+        domain.user_actions = domain._custom_actions
```

Comment:
This might be nitpicking, but shouldn't protected attributes like `_custom_actions` not be accessible outside the class? The alternative would be the more convoluted way:
```
custom_actions = domain.as_dict.get(KEY_ACTIONS)
```

Reply:
No, you're right I shouldn't be using this. Same with `_combine_user_with_default_actions`. Maybe the best solution is to add it to the domain 😬 . (And the even better solution would be to let the domain remember which parameters where used to instantiate it, I think, but I won't touch that now 😄 ).

Reply:
Something to add to the ideas doc 👍🏼 In the meantime, wouldn't `set(domain.action_names_or_texts).difference(responses)` work as a way to remove responses without using `_combine_user_with_default_actions`? Would probably need to be bumped a line or two before you assign to `responses` the empty dict.

Reply:
Yep, same with `user_actions` - but maybe it's really better to just move this to Domain so we don't miss that during the disentangling work (especially because in the current form it is very clearly 1:1 what the initialisation would do) and then we don't need to know/foresee how `action_names_or_texts` and the other things (will) look like ?

Reply:
By moving to Domain, you mean having methods in Domain that disentangle responses from `user_actions` and `action_names_or_texts`?

Reply:
I mean adding a "def recreate_without_responses(self)"

Reply:
Oh right, yes, that would be the cleanest! @wochinge would this be out of scope for this PR or fine to action it here?

Reply:
yep, I'd prefer that, too 👍🏻 Wouldn't a quick way to dump responses also be to

```python
serialized = domain.as_dict()
del serialized["responses"]
return Domain.from_dict(serialized)
```

Reply:
Ohhhh, perfect, yes :) Had expected `as_dict` to return all/most attributes as a dict but that doesn't contain any of the funny mixed lists